lld/ELF/InputFiles.h

```diff
@@ -360,10 +360,22 @@ class SharedFile : public ELFFileBase {
   // parsed. Only filled for `--no-allow-shlib-undefined`.
   SmallVector<Symbol *, 0> requiredSymbols;
 
+  template <typename ELFT> typename ELFT::PhdrRange getELFPhdrs() const {
+    return typename ELFT::PhdrRange(
+        reinterpret_cast<const typename ELFT::Phdr *>(elfPhdrs), numElfPhdrs);
+  }
+
+protected:
+  const void *elfPhdrs = nullptr;
```

Comment:
getELFPhdrs is only needed once in `parse`. Do not cache them. `getELFPhdrs` can be deleted as well.